### ISSUE TITLE
Multiclass Training - WARNING: amalgamation/../src/learner.cc:541

### ISSUE THREAD

**Issue by KevinJChalifoux** [dmlc/xgboost#6850]
Hi,

I've been using this package for over a year now and I really like it. Earlier in the year, I made an update and everything still worked fine until the following update I made about a week ago. Calling the "train_model" function below, on the features (data) and errors given in the files at the end of this message, makes it unable to make predictions (using the function stats::predict below) and I now obtain this warning message:

* 
WARNING: amalgamation/../src/learner.cc:541: 
Parameters: { silent } might not be used.

  This may not be accurate due to some parameters are only used in language bindings but
  passed down to XGBoost core.  Or some parameters are not used but slip through this
  verification. Please open an issue if you find above cases.
*

I use these functions in a multiclassification problem using the classification errors and the features as input. If the issue is due to a misuse of your functions, I'd appreciate your corrections to my code.

Thank you, in advance, for your kind help :) 

# Train Function
train_model <- function (data, errors) {
dtrain <- xgboost::xgb.DMatrix(data)
attr(dtrain, "errors") <- errors

param <- list(max_depth=14, eta=0.575188, nthread = 3, silent=1,
                objective=error_softmax_obj,
                num_class=ncol(errors),
                subsample=0.9161483,
                colsample_bytree=0.7670739, eval_metric = "merror"
  )

return(xgboost::xgb.train(param, dtrain, 94))
}

# Prediction line
pred <- stats::predict(xgb_returned_model, new_data, outputmargin = TRUE, reshape=TRUE)

# Folder containing sample of features and errors
[issue_xgboost.zip](https://github.com/dmlc/xgboost/files/6305841/issue_xgboost.zip)

**Comment by trivialfis** [comment]
Thanks for opening an issue, the `silent` parameter is replaced by `verbosity`.  Setting it to 0 means `silent`.

**Comment by trivialfis** [comment]
Could you please share you `error_softmax_obj`?

**Comment by KevinJChalifoux** [comment]
Hi. 

Here is the error_softmax_obj function I used, inspired by the FFORMA model used in the M4 competition.

# Softmax Transformation
softmax_transform <- function(x) {
  exp(x) / sum(exp(x))
}

# Error Softmax Function
error_softmax_obj <- function(preds, dtrain) {
  labels <- xgboost::getinfo(dtrain, "label")
  errors <- attr(dtrain, "errors")

  nb_fcters <- ncol(errors)
  nb_series <- nrow(errors)

  predictions <- matrix(ncol = nb_fcters, nrow = nb_series)
  for(i in 1:nb_fcters) {
    predictions[, i] <- preds[((i-1)*nb_series +1):(i*nb_series)]
  }

  predictions <- exp(predictions)
  sp <- rowSums(predictions)
  predictions <- predictions / replicate(ncol(predictions), sp)
  rowsumerrors <- replicate( ncol(predictions), rowSums(predictions * errors))

  grad <- predictions*(errors - rowsumerrors)
  hess <- errors*predictions*(1.0-predictions) - grad*predictions
  #print(grad)
  #print(hess)
  return(list(grad = t(grad), hess = t(hess)))
}

**Comment by KevinJChalifoux** [comment]
When I replace silent = 1 by verbosity = 0, I obtain the following error calling the xgboost function, for the same data: 

Error in xgb.get.DMatrix(data, label, missing, weight) : 
  xgboost: invalid input data
In addition: Warning message:
In xgb.get.DMatrix(data, label, missing, weight) :
  xgboost: label will be ignored.

**Comment by trivialfis** [comment]
Hi, sorry for the slow reply.  I will be back from vacation tomorrow.
